research/quantifying-data-leakage/src/nvflare_gradinv/utils/create_job_config.py

```diff
@@ -0,0 +1,207 @@
+    if not os.path.isdir(os.path.dirname(out_server_config_file)):
+        os.makedirs(os.path.dirname(out_server_config_file))
+
+    save_config(out_server_config_file, server_config)
+    print(
+        f"Created {out_server_config_file} to use min_clients={server_config['min_clients']} and num_rounds={server_config['num_rounds']}"
+    )
+
+
+def create_client_config(client_config_file, args, client, bs):
+    client_config = load_config(client_config_file)
+
+    client_config["DATA_ROOT"] = args.data_root
+    client_config["DATASET_JSON"] = args.dataset_json_prefix + f"{client}.json"
+
+    client_config["BATCH_SIZE"] = bs
+
+    client_config["SIGMA0"] = args.sigma0
+
+    if args.prior_file and "PRIOR_FILE" in client_config:
+        client_config["PRIOR_FILE"] = args.prior_file
+
+    out_client_config_file = os.path.join(args.job_config, f"site-{client}", "config", "config_fed_client.json")
+    if not os.path.isdir(os.path.dirname(out_client_config_file)):
+        os.makedirs(os.path.dirname(out_client_config_file))
+
+    save_config(out_client_config_file, client_config)
+
+    print(
+        f"Created {out_client_config_file} to use DATA_ROOT={client_config['DATA_ROOT']}, DATASET_JSON={client_config['DATASET_JSON']}, and SIGMA0={client_config['SIGMA0']}"
+    )
+    if "PRIOR_FILE" in client_config:
+        print(f"Used prior_file {client_config['PRIOR_FILE']}")
+
+
+def create_gradinv_config(inv_config_file, args, client):
+    inv_config = load_config(inv_config_file)
+
+    inv_config["img_prior"] = args.prior_file
+
+    out_inv_config_file = os.path.join(args.job_config, f"site-{client}", "config", "config_inversion.json")
+    if not os.path.isdir(os.path.dirname(out_inv_config_file)):
+        os.makedirs(os.path.dirname(out_inv_config_file))
+
+    save_config(out_inv_config_file, inv_config)
+
+    print(f"Created {out_inv_config_file} to use prior_file {inv_config['img_prior']}")
+
+
+def create_meta_config(clients, meta_config_file, args):
+    meta_config = load_config(meta_config_file)
+
+    deploy_map = {"server": ["server"]}
+    for client in clients:
+        deploy_map.update({f"site-{client}": [f"site-{client}"]})
+
+    meta_config["deploy_map"] = deploy_map
+
+    out_meta_config_file = os.path.join(args.job_config, "meta.json")
+    if not os.path.isdir(os.path.dirname(out_meta_config_file)):
+        os.makedirs(os.path.dirname(out_meta_config_file))
+
+    save_config(out_meta_config_file, meta_config)
+
+    print(f"Created {out_meta_config_file}")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--app_folder",
+        type=str,
+        default="app_template",
+        help="Folder containing app config files in JSON format.",
+    )
+    parser.add_argument(
+        "--job_config",
+        type=str,
+        help="Output folder containing the job config.",
+    )
```

Comment:
Would "--output" or "--destination" make it more clear?
From the argument name "job_config" I would at first think you are reading from here.